src/main/scala/com/fulcrumgenomics/bam/ZipperBams.scala

```diff
@@ -0,0 +1,191 @@
+  def merge(unmapped: Template, mapped: Template, tagInfo: TagInfo): Template = {
+    // Fix mate info first
+    mapped.fixMateInfo()
+
+    // Remove tags from the mapped records
+    for (rec <- mapped.allReads; tag <- tagInfo.remove) rec.remove(tag)
+
+    // Copy tags over from the unmapped record
+    for (u <- unmapped.primaryReads) {
+      // Get all the mapped records for the unmapped record
+      val ms = if (u.unpaired || u.firstOfPair) mapped.allR1s.toSeq else mapped.allR1s.toSeq
+      // Copy over tags on any positive strand reads first
+      ms.iterator.filter(_.positiveStrand).foreach(m => copyTags(u, m))
+
+      // Then reverse complement the unmapped read if we need to
+      if (tagInfo.hasRevsOrRevcomps && ms.exists(_.negativeStrand)) {
+        u.asSam.reverseComplement(tagInfo.setToRevcomp, tagInfo.setToReverse, true)
+      }
+
+      // And deal with the negative strand mappings
+      ms.iterator.filter(_.negativeStrand).foreach(m => copyTags(u, m))
+    }
+
+    mapped
+  }
+
+  /** Copies all tags from read to another. */
+  def copyTags(src: SamRecord, dest: SamRecord): Unit = {
+    src.attributes.foreach { case (tag, value) => dest(tag) = value }
+  }
+}
+
+@clp(group=ClpGroups.SamOrBam, description=
+  """
+    |Zips together an unmapped and mapped BAM to transfer metadata into the output BAM.
+    |
+    |Both the unmapped and mapped BAMs _must_ be a) queryname sorted or grouped (i.e. all records with the same
+    |name are grouped together in the file), and b) have the same ordering of querynames.  If either of these are
+    |violated the output is undefined!
+    |
+    |By default the mapped BAM is read from stdin and the output BAM is written to stdout. This can be changed
+    |using the `--input/-i` and `--output/-o` options.
+    |
+    |By default the output BAM file is emitted in the same order as the input BAMs.  This can be overridden
+    |using the `--sort` option, though in practice it may be faster to do the following:
+    |
+    |  `fgbio --compression 0 ZipperBams -i mapped.bam -u unmapped.bam -r ref.fa | samtools sort -@ 16`
+    |
+    |
+  """)
+class ZipperBams
+( @arg(flag='i', doc="Mapped SAM or BAM.") val input: PathToBam = Io.StdIn,
+  @arg(flag='u', doc="Unmapped SAM or BAM.") val unmapped: PathToBam,
+  @arg(flag='r', doc="Path to the reference used in alignment.") val ref: PathToFasta,
+  @arg(flag='o', doc="Output SAM or BAM file.") val output: PathToBam = Io.StdOut,
+  @arg(doc="Tags to remove from the mapped BAM records.", minElements=0) val tagsToRemove: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse on reads mapped to the negative strand.", minElements=0) val tagsToReverse: IndexedSeq[String] = IndexedSeq.empty,
+  @arg(doc="Set of optional tags to reverse complement on reads mapped to the negative strand.", minElements=0) val tagsToRevcomp: IndexedSeq[String] = IndexedSeq.empty,
```

Comment:
Any harm in putting in the usage our recommended values for duplex sequencing?

Reply:
I was thinking actually that since tags have to be two characters we could also allow named sets, so you could do --reverse Duplex and --revcop Duplex and it would put in all the relevant tags?